### PR TITLE
Minor updates in Pytorch interface

### PR DESCRIPTION
**Pull request by olivierverdier**
Mostly due to interface changes in Pytorch.

### CODE REVIEW

File: odl/contrib/torch/operator.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def forward(self, input):
             self.save_for_backward(input)
 
         # TODO: use GPU memory directly if possible
-        input_arr = input.cpu().numpy()
+        input_arr = input.detach().numpy()
```
Comment (kohr-h) [inline odl/contrib/torch/operator.py]: If `input` is on the GPU, this fails. We need both `detach()` and `cpu()`.

Comment (olivierverdier) [reply odl/contrib/torch/operator.py]: Do you mean `input.cpu().detach().numpy()`?

Comment (kohr-h) [reply odl/contrib/torch/operator.py]: Yes, or `input.detach().cpu().numpy()`, whatever you like better. Perhaps a little helper `to_numpy` would make that less ugly.

File: odl/contrib/torch/operator.py [hunk 2]
```diff
@@ -16,6 +16,12 @@
 from __future__ import division
 import numpy as np
 import torch
+import warnings
+from pkg_resources import parse_version
+
+if parse_version(torch.__version__) < parse_version('0.4'):
+    warnings.warn("This interface is designed to work"
```
Comment (adler-j) [inline odl/contrib/torch/operator.py]: the strings will be concatenated like "workwith", you need to add a space after "work"
